FrontEnd/src/main/java/resourceDisplay/AssetsRepository.java

```diff
@@ -0,0 +1,313 @@
+package resourceDisplay;
+
+import com.google.cloud.spanner.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This Class does all DB access
+ */
+public class AssetsRepository {
+    /**
+     * This is just for testing the function that returns assets by type
+     * @return a list of TestAsset objects
+     */
+    public List<TestAsset> executeAllAssetsQuery(Statement statement, DatabaseClient dbClient){
+       List<TestAsset> allAssets = new ArrayList<>();
+        try (ResultSet resultSet = dbClient.singleUse().executeQuery(statement)) {
+            while (resultSet.next()) {
+                TestAsset asset = new TestAsset();
+                if (!resultSet.isNull("assetName")){
+                    asset.setName(resultSet.getString("assetName"));
+                }
+                if (!resultSet.isNull("kind")){
+                    asset.setKind(resultSet.getString("kind"));
+                }
+                if (!resultSet.isNull("location")){
+                    asset.setZone(resultSet.getString("location"));
+                }
+                if (!resultSet.isNull("creationTime")){
+                    asset.setCreationTime(resultSet.getTimestamp("creationTime"));
+                }
+                if (!resultSet.isNull("status")){
+                    asset.setStatus(resultSet.getString("status"));
+                }
+                allAssets.add(asset);
+            }
+            return allAssets;
+        }
+    }
+    /**
+     * This is used to return all the distinct statuses in a project/workspace (at the moment workspaceID)
+     * @param dbClient - A client for connection to the DB
+     * @param workspaceId - The ID of the wanted workspace
+     * @return a list of strings holding the different statuses in a project/workspace
+     */
+    public List<String> getStatusList(DatabaseClient dbClient, String workspaceId) {
+        List<String> statusList = new ArrayList<>();
+        Statement statement =
+                Statement.newBuilder(
+                        "SELECT DISTINCT status FROM Main_Assets "
+                                + "WHERE workspaceId = @workspaceId")
+                        .bind("workspaceId")
+                        .to(workspaceId)
+                        .build();
+        try (ResultSet resultSet = dbClient.singleUse().executeQuery(statement)) {
```

Comment:
Two things:
a. Is it possible to express this list as a Java streams operation? (stream(), fileter(), etc)
b. Will it make sense to sort this?
(also in a few other places below)